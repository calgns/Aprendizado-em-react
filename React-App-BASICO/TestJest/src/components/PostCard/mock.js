
export const postCardPropsMock = {
  title: 'title 1',
  body: 'body 1',
  id: 1,
  // cover: "https://placehold.co/600x400",
  cover: "https://placeimg.com/640/480/any",
}