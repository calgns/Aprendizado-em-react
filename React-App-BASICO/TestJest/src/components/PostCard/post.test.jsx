import { render, screen } from "@testing-library/react";
import { PostCard } from ".";
import { postCardPropsMock } from "./mock";

const props = postCardPropsMock;

describe("<PostCard />", () => {
  it("should render PostCard correctly", () => {
    // const {debug} = render(<PostCard {...mock} />);
    render(<PostCard {...props} />);

    expect(screen.getByAltText(/title 1/i)).toHaveAttribute(
      "src",
      "https://placeimg.com/640/480/any"
    );
    expect(screen.getByRole("img", { name: props.title })).toBeInTheDocument(); // name: 'title 1' | name: /title/i
    expect(
      screen.getByRole("heading", { name: /title 1/i })
    ).toBeInTheDocument();
    expect(screen.getByText("body 1")).toBeInTheDocument();
    // debug();
  });

  it("Should match snapshot", () => {
    const { container } = render(<PostCard {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
