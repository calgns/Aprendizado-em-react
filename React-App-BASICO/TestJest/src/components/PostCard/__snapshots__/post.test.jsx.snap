// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> Should match snapshot 1`] = `
<div
  class="post-card"
>
  <img
    alt="title 1"
    src="https://placeimg.com/640/480/any"
  />
  <div
    class="card-text"
  >
    <h2>
      title 1
       
      1
    </h2>
    <hr />
    <p>
      body 1
    </p>
  </div>
</div>
`;
